backend/substrapp/tasks/tasks.py

```diff
@@ -209,14 +208,24 @@ def prepare_traintuple_input_models(directory, tuple_):
 
     models = []
     for input_model in input_models:
-        proc = Thread(target=fetch_model,
-                      args=(TRAINTUPLE_TYPE, authorized_types, input_model, directory))
+        proc = ExceptionThread(target=fetch_model,
+                               args=(TRAINTUPLE_TYPE, authorized_types, input_model, directory))
         models.append(proc)
         proc.start()
 
     for proc in models:
         proc.join()
 
+    exceptions = []
+
+    for proc in models:
+        if hasattr(proc, "_exception"):
+            exceptions.append(proc._exception)
+            logger.exception(proc._exception)
+    else:
```

Comment:
This made me discover (again) the `for...else` syntax. Since we don't have a break statement in the loop isn't the else clause unecessary here? 
Unless this is an existing code convention, in which case nevermind.

Reply:
No particular reason, I just do that because it concerns the same idea (looping, then raising)